app/models/course/assessment/question/programming.rb

```diff
@@ -174,10 +192,10 @@ def process_package
     elsif should_evaluate_package
       # For non-autograded questions, the attachment is not present
       evaluate_package if attachment
-    elsif is_codaveri_changed? || live_feedback_enabled_changed?
-      # Only when is_codaveri changed (no other setting), we recreate the codaveri
-      # problem to avoid attachment recreation and answers regrading
-      create_codaveri_problem if attachment
+    elsif (is_codaveri_changed? && is_codaveri?) || (live_feedback_enabled_changed? && live_feedback_enabled?)
+      # changes in other part of question (title and description) also needs to
```

Comment:
those title and description is just an example of what might get updated in the period of not being a Codaveri problem. The main point is that for any changes, we need to sync it with Codaveri.

To prevent confusion, I have removed those examples, and only write `changes in other part of question also needs to be synced to Codaveri for precise feedback`

Reply:
> any changes

Not all changes need to be synced back to Codaveri

Reply:
with Codaveri, almost all parts of each programming question needs to be synced with Codaveri

- title
- description
- language (name and version)
- templates (solutions)
- all test cases (public, private, evaluation) -> each has expression, expected, and hint

introducing logic to check if any one of those fields have changed will complicate the logic and makes the code very messy, hence it should be in our best interest that everytime we update the programming question, we mark it as not synced so that later when we really need to sync (getting live feedback, evaluating or generating codaveri feedback) we sync it properly and on-demand